Celestial/FigCaptureSourceFormat.h

```diff
@@ -0,0 +1,12 @@
+#import <Foundation/Foundation.h>
+
+API_AVAILABLE(ios(8.0))
+@interface FigCaptureSourceFormat : NSObject
+
+@property (readonly) NSArray *AVCaptureSessionPresets API_DEPRECATED("Removed in iOS 11", ios(8.0, 11.0));
```

Comment:
Not necessarily for this PR, but more as a general question. Do we know the type of the array items? I think specifying `NSArray <Type> *` is useful but iirc it’s something supported since iOS 4 or 6 times so I’m unsure how much we could add this type info.

Reply:
regarding generics in general, there are currently generics used in this repo: https://github.com/theos/headers/commit/a4e47e7399ab73b611c02f6fd850a37549bb2037

`AVBase.h` used to declare
```objc
// Generics

// Use when declaring a variable of a generic type
#if __has_feature(objc_generics)
    #define AV_GENERIC(BASETYPE, ...) BASETYPE<__VA_ARGS__>
#else
    #define AV_GENERIC(BASETYPE, ...) BASETYPE
#endif
```